compiler/rustc_error_messages/locales/en-US/ast_passes.ftl

```diff
@@ -89,3 +92,222 @@ ast_passes_fn_without_body =
+
+ast_passes_foreign_item_non_ascii =
+    items in `extern` blocks cannot use non-ascii identifiers
+    .extern_block_label = in this `extern` block
+    .note = this limitation may be lifted in the future; see issue #83942 <https://github.com/rust-lang/rust/issues/83942> for more information
+
+ast_passes_forbidden_c_var_args =
+    only foreign or `unsafe extern "C"` functions may be C-variadic
+
+ast_passes_unnamed_assoc_const =
+    `const` items in this context need a name
+    .label = `_` is not a valid name for this `const` item
+
+ast_passes_nomangle_item_non_ascii =
+    `#[no_mangle]` requires ASCII identifier
+
+ast_passes_mod_file_item_non_ascii =
+    trying to load file for module `{$name}` with non-ascii identifier name
+    .help = consider using `#[path]` attribute to specify filesystem path
+
+ast_passes_auto_trait_with_generic_param =
+    auto traits cannot have generic parameters
+    .ident_label = auto trait cannot have generic parameters
+    .suggestion = remove the parameters
+
+ast_passes_auto_trait_with_super_trait_or_where_clause =
+    auto traits cannot have super traits or lifetime bounds
+    .ident_label = auto trait cannot have super traits or lifetime bounds
+    .suggestion = remove the super traits or lifetime bounds
+
+ast_passes_auto_trait_with_assoc_item =
+    auto traits cannot have associated items
+    .suggestion = remove these associated items
+    .ident_label = auto trait cannot have associated items
+
+ast_passes_generic_arg_after_constraint =
+    generic arguments must come before the first constraint
+    .first_constraint_label = { $constraints_len ->
+        [one] constraint
+       *[other] constraints
+    }
+    .last_arg_label = { $args_len ->
+        [one] generic argument
+       *[other] generic arguments
+    }
+    .constraints_label = {""}
+    .args_label = {""}
+    .suggestion = move the { $constraints_len ->
+        [one] constraint
+       *[other] constraints
+    } after the { $args_len ->
+        [one] generic argument
+       *[other] generic arguments
+    }
+
+ast_passes_fn_ptr_ty_with_pat =
+    patterns aren't allowed in function pointer types
+
+ast_passes_multiple_explicit_lifetime_bound =
+    only a single explicit lifetime bound is permitted
+
+ast_passes_impl_trait_ty_in_path_param =
+    `impl Trait` is not allowed in path parameters
+
+ast_passes_impl_trait_ty_nested =
+    nested `impl Trait` is not allowed
+    .outer_label = outer `impl Trait`
+    .nested_label = nested `impl Trait` here
+
+ast_passes_impl_trait_ty_without_trait_bound =
+    at least one trait must be specified
+
+ast_passes_deprecated_extern_missing_abi =
+    extern declarations without an explicit ABI are deprecated
+
+ast_passes_generic_param_wrong_order =
+    { $param_kind ->
+        [lifetime] lifetime
+       *[type_or_const] type and const
+    } parameters must be declared prior to { $max_param_kind ->
+        [lifetime] lifetime
+       *[type_or_const] type and const
```

Comment:
Can these be split into multiple Fluent messages to avoid pattern matching on strings?